packages/compiler/testing/test-server-host.ts

```diff
@@ -81,17 +81,34 @@ export async function createTestServerHost(options?: TestHostOptions) {
 }
 
 /**
- * Takes source code with a cursor position indicated by `┆`, removes the
- * `┆` and returns the source without the `┆` and the numeric cursor
- * position.
+ * Takes source code with a cursor position indicated by the given marker
+ * ("┆" by default), and returns the source without the marker along with
+ * the cursor position.
  */
-export function extractCursor(sourceWithCursor: string): { source: string; pos: number } {
-  const pos = sourceWithCursor.indexOf("┆");
-  ok(pos >= 0, "no cursor found");
-  const source = sourceWithCursor.replace("┆", "");
+export function extractCursor(
+  sourceWithCursor: string,
+  marker = "┆"
+): { source: string; pos: number } {
+  const pos = sourceWithCursor.indexOf(marker);
+  ok(pos >= 0, "marker not found");
+  const source = sourceWithCursor.replace(marker, "");
   return { source, pos };
 }
 
+/**
+ * Takes source code with start and end positions indicated by given marker
+ * ("~~~" by default) and returns the source without the markers along with
+ * the start and end positions.
+ */
+export function extractSquiggles(
```

Comment:
oh haha made the same change in https://github.com/microsoft/cadl/pull/749

Reply:
:)

Reply:
Haha, I promised you I'd do it if I needed it a second time last time around, so making good on it.